### PR TITLE
Downloads: add switch for saf/legacy file picker

### LINKED ISSUE #2498: crash when trying to download
when try choose / folder where download content apps auto-close and close all activity 

whem try download content i can,t download 
start download work but download speed is very slow (500 ko) not due to my network

### LINKED ISSUE #2504: Can't enable "Documents" for 0.17.0 on Android 5.x
- [x] I carefully read the [contribution guidelines](https://github.com/TeamNewPipe/NewPipe/blob/HEAD/.github/CONTRIBUTING.md) and agree to them.
- [x] I checked if the issue/feature exists in the latest version.
- [ ] I did use the [incredible bugreport to markdown converter](https://teamnewpipe.github.io/CrashReportToMarkdown/) to paste bug reports.

After upgrading from `0.16.2` to `0.17.0` NewPipe start asking for:

> Turn on Documents in Settings > Installed apps first

pic.1

pic.2

But when I open permissions of NewPipe in "Installed apps" I see that there NO "Documents".

pic.3

How should I allow downloading for `0.17.0`?

Android: `5.1.1` (Redmi 3)

### PR DESCRIPTION
**Pull request by kapodamy**
Some ROMs are not [CTS](https://source.android.com/compatibility/cts/) compliant, in these cases the Storage Access Framework is not fully implemented or functional.
This PR checks whatever is SAF available, otherwise, the legacy API (Java I/O) will be used

changes:
* restore request permissions popups
* use legacy file/folder picker if necessary

fixes #2498 and fixes #2504 

- [ ✔ ] I carefully read the [contribution guidelines](https://github.com/TeamNewPipe/NewPipe/blob/HEAD/.github/CONTRIBUTING.md) and agree to them.

### CODE REVIEW

**Comment by TobiGr** [comment]
@kapodamy Thanks!
Here is an APK for testing:
[app-debug.apk.zip](https://github.com/TeamNewPipe/NewPipe/files/3504651/app-debug.apk.zip)

**Comment by theScrabi** [comment]
Worx

**Comment by kapodamy** [comment]
last try (before add an "manual switch") in commit https://github.com/TeamNewPipe/NewPipe/pull/2521/commits/8f13a7ec973c7f5180b03ef7867dc608f726cc8a i added another check

File: app/src/main/res/values/strings.xml [hunk 1]
```diff
@@ -556,6 +556,8 @@
 
     <string name="downloads_storage_ask_title">Ask where to download</string>
     <string name="downloads_storage_ask_summary">You will be asked where to save each download</string>
-    <string name="downloads_storage_ask_summary_kitkat">You will be asked where to save each download.\nEnable this option if you want download to the external SD Card</string>
+    <string name="downloads_storage_ask_summary_kitkat">You will be asked where to save each download.\nEnable this option with SAF if you want download to the external SD Card</string>
```
Comment (TobiGr) [inline app/src/main/res/values/strings.xml]:
```suggestion
    <string name="downloads_storage_ask_summary_kitkat">You will be asked where to save each download.\nChoose SAF if you want to download to an external SD card</string>
```

File: app/src/main/res/values/strings.xml [hunk 2]
```diff
@@ -556,6 +556,8 @@
...
+    <string name="downloads_storage_use_saf_title">Use SAF</string>
+    <string name="downloads_storage_use_saf_summary">The Storage Access Framework allow download to the external SD Card.\nNote: some devices are not compatible</string>
```
Comment (TobiGr) [inline app/src/main/res/values/strings.xml]:
```suggestion
    <string name="downloads_storage_use_saf_summary">The Storage Access Framework allows downloads to the external SD card.\nNote: some devices are not compatible</string>
```

**Pull request merged**